### PR TITLE
fix(check-monitored-trip): advance to next trip day when itinerary is no longer valid

### PR DESCRIPTION
**Pull request by josh-willis-arcadis**
### Checklist

- [ ] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [ ] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [ ] The description lists all applicable issues this PR seeks to resolve
- [ ] The description lists any configuration setting(s) that differ from the default settings
- [ ] All tests and CI builds passing
- [ ] The description lists all relevant PRs included in this release _(remove this if not merging to master)_

### Description

After submitting a trip with certain monitored days, the next trip can be shown on a day that is not monitored. Adds a check on the previous matching itinerary against days the trip should be monitored.

### CODE REVIEW

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 1]
```diff
@@ -29,14 +29,7 @@
...
 import java.time.temporal.ChronoUnit;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Date;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: We use explict imports. Can you revert this please. There is also a setting in Intellij (assuming you are using this) that prevents this.

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 2]
```diff
@@ -771,6 +764,33 @@ public boolean shouldSkipMonitoredTripCheck(boolean persist) throws Exception {
         return true;
     }
 
+    // Check if previous matching itinerary day is still valid
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: Please change this to a method comment. Start with `/**` and end with `*/`

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 3]
```diff
@@ -771,6 +764,33 @@ public boolean shouldSkipMonitoredTripCheck(boolean persist) throws Exception {
...
+    private boolean isPrevMatchingItineraryDayValid() {
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(previousMatchingItinerary.startTime);
+
+        int dayOfWeek = calendar.get(Calendar.DAY_OF_WEEK);
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: Nit. You could in-line this variable.

**Review by br648 — APPROVED** [review]
A few minors, but apart from them, all good.

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java (hunk 1 above)
Comment (josh-willis-arcadis) [reply src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: Just updated my IDE settings. Thanks for pointing this out!

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java (hunk 2 above)
Comment (josh-willis-arcadis) [reply src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: addressed in [3bd3cb1](https://github.com/ibi-group/otp-middleware/pull/284/commits/3bd3cb1672e02da0b1ed37af5f9e644c42215403)

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java (hunk 3 above)
Comment (josh-willis-arcadis) [reply src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: addressed in [3bd3cb1](https://github.com/ibi-group/otp-middleware/pull/284/commits/3bd3cb1672e02da0b1ed37af5f9e644c42215403)

**Review by miles-grant-ibigroup — APPROVED** [review]
I'm a little confused as to what's going on here but I trust you

**Pull request merged**